log/log.go

```diff
@@ -2,158 +2,57 @@ package log
+func NewLogger(component string, keyValuePairs ...interface{}) logr.Logger {
+	return NewLoggerWithOptions(component, nil, keyValuePairs...)
 }
 
-// InitWithOptions inits the logger with the provided opts
-func InitWithOptions(component string, opts []Option, keyValuePairs ...interface{}) {
-	mtx.Lock()
-	defer mtx.Unlock()
-
-	ll := NewLogger(component, defaultOutput, 0, JSONEncoder{}, keyValuePairs...)
+// NewLoggerWithOptions creates a logger with the provided opts and key value pairs
+func NewLoggerWithOptions(component string, opts []Option, keyValuePairs ...interface{}) logr.Logger {
+	s := NewLogSink(component, defaultOutput, 0, JSONEncoder{}, keyValuePairs...)
 
 	for _, opt := range opts {
-		opt(ll)
+		opt(s)
 	}
 
-	// don't lock because we already have a lock
-	useLogger(ll)
-}
-
-// MustInitWithOptions calls InitWithOptions and panics if an error is returned
-func MustInitWithOptions(component string, opts []Option, keyValuePairs ...interface{}) {
-	InitWithOptions(component, opts, keyValuePairs...)
-}
-
-// GetLogger returns the root logger used for logging
-func GetLogger() logr.Logger {
-	return logger
-}
-
-// UseLogger bypasses the requirement for Init and sets the logger to l
-func UseLogger(l logr.Logger) {
-	mtx.Lock()
-	defer mtx.Unlock()
-	useLogger(l)
-}
-
-// useLogger sets the logger to l without mtx.Lock()
-// To use mtx.Lock see UseLogger
-func useLogger(l logr.Logger) {
-	logger = l
-}
-
-// Info logs a non-error message with the given key/value pairs as context.
-//
-// The msg argument should be used to add some constant description to
-// the log line.  The key/value pairs can then be used to add additional
-// variable information.  The key/value pairs should alternate string
-// keys and arbitrary values.
-func Info(msg string, keysAndValues ...interface{}) {
-	mtx.RLock()
-	defer mtx.RUnlock()
-	logger.Info(msg, keysAndValues...)
+	return logr.New(s)
 }
 
-// Error logs an error, with the given message and key/value pairs as context.
-// It functions similarly to calling Info with the "error" named value, but may
-// have unique behavior, and should be preferred for logging errors (see the
-// package documentations for more information).
-//
-// The msg field should be used to add context to any underlying error,
-// while the err field should be used to attach the actual error that
-// triggered this log line, if present.
-func Error(err error, msg string, keysAndValues ...interface{}) {
-	mtx.RLock()
-	defer mtx.RUnlock()
-	logger.Error(err, msg, keysAndValues...)
-}
-
-// WithValues adds some key-value pairs of context to a logger.
-// See Info for documentation on how key/value pairs work.
-func WithValues(keysAndValues ...interface{}) logr.Logger {
-	mtx.RLock()
-	defer mtx.RUnlock()
-	return logger.WithValues(keysAndValues...)
-}
+// GetSink return the LogSink converted as a Sink object. It returns an
+// error if it cannot convert it.
+func GetSink(l logr.Logger) (*Sink, error) {
```

Comment:
Do we still need `GetSink` / `MustGetSink`?

Reply:
Probably not. See the comment to your question around making `Sink` internal for reasoning.